src/mujincontrollerclient.cpp

```diff
@@ -89,7 +91,45 @@ bool ControllerClientInfo::operator==(const ControllerClientInfo &rhs) const
            unixEndpoint == rhs.unixEndpoint;
 }
 
-std::string ControllerClientInfo::GetURL(bool bIncludeNamePassword) const
+std::string ResolveIpFromHostIfUnique(const std::string& host)
+{
+    struct addrinfo hints = {}, *addrs;
+
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_INET;  // IPv4 only
+    hints.ai_socktype = SOCK_STREAM;
+
+    int status = getaddrinfo(host.c_str(), nullptr, &hints, &addrs);
+    if (status != 0) {
+        std::stringstream ss;
+        ss << "host resolution failed for host " << host << ": " << gai_strerror(status);
+        MUJIN_LOG_ERROR(ss.str());
```

Comment:
@wkentaro i think you can put `"host resolution failed for host " << host << ": " << gai_strerror(status)` as an argument.